.github/workflows/gcp-docker.yml

```diff
@@ -0,0 +1,99 @@
+name: GCP Docker Build
+
+on:
+  workflow_dispatch:
+    inputs:
+      push:
+        description: "Push the image to the registry"
+        required: false
+        default: false
+        type: boolean
+  push:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  packages: write
+  id-token: write
+
+jobs:
+  base:
+    name: Base image
+    uses: ./.github/workflows/reusable-gcp-docker-build-push.yml
+    with:
+      image_name: base
+      dockerfile: ./Dockerfile
+      context: .
+      push: ${{ github.event.inputs.push || github.ref == 'refs/heads/main' }}
```

Comment:
If these are all scoped to `refs/heads/main` individually could this not just be only triggered on the main branch?

Reply:
yes though I'm actually using it to control the push specifically so the build itself can still be tested for non-`main` branches.
however, the `builder` and `runtime` could be disabled and just use the `monolith` as the "canary"?

Reply:
see comments below

Reply:
@wfaithfull updated pipeline running now if you want to look?

Reply:
Ah fantastic, yes, I see. It is skipping the pushes